src/scenario/scenario.cpp

```diff
@@ -249,7 +249,14 @@ void cScenario::import_legacy(legacy::scenario_data_type& old){
 	horses.resize(30);
 	for(short i = 0; i < 30; i++) {
 		boats[i].import_legacy(old.scen_boats[i]);
+		// in the .exs scenario, the boats in town seem to be defined with exists=false
```

Comment:
This seems suspicious. What's the bug that this is even trying to fix? I think the issue might be rather that `exists` isn't relevant in a scenario file and is supposed to be overwritten when the scenario is loaded for a party, but I'm not sure of the details.

This could also possibly be a symptom of an issue with the legacy structs? They're supposed to be defined without padding, but is it possible the compiler is ignoring the pack pragma?

Reply:
Yes, you can probably skip this patch (ie. _exists_ was not relevant in a scenario file, but there exists some code in start_town_mode which does that).

**Notes**:  I guess I was trying to figure out why some boats/horses were disappearing outside but there is another patch to fix that.

Reply:
I'm guessing the other patch in question is this one?

3cc550b4d9b7aa9d7909bbb25022907df91d9810

Reply:
Yes.